lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -950,9 +760,15 @@ def _from_file(self, root_path: Path, file_path: Path) -> List[dict]:
                         alert_rule["labels"] = {}
 
                     if self.topology:
-                        alert_rule["labels"].update(self.topology.as_promql_label_dict())
+                        alert_rule["labels"].update(self.topology.label_matcher_dict)
                         # insert juju topology filters into a prometheus alert rule
-                        alert_rule["expr"] = self.topology.render(alert_rule["expr"])
+                        # logql doesn't like empty matchers, so add a job matcher which hits
+                        # any string as a "wildcard" which the topology labels will
+                        # filter down
+                        alert_rule["expr"] = self.tool.inject_label_matchers(
+                            re.sub(r"%%juju_topology%%(,?)", r'job=".+"\1', alert_rule["expr"]),
+                            self.topology.label_matcher_dict,
+                        )
```

Comment:
It's required here (which is how it wound in Prometheus to stay in sync).

The difference in this case is that LogQL _cannot_ have empty matchers, or the expression will fail to parse, which is why a "wildcard" `job=".+"` matcher is injected. If a comma which was present is omitted, in, say, `{%%juju_topology%%, juju_unit="foo"}`, it would become `{job=".+" juju_unit="foo"}`, which wouldn't parse.